test/functional/kernel/nested-loop-view-types/test-kernel-nested-loop-view.cpp.in

```diff
@@ -0,0 +1,332 @@
+        >
+      >
+    >
+  >,
+
+  RAJA::KernelPolicy<
+    RAJA::statement::Collapse<RAJA::seq_exec, RAJA::ArgList<0, 1, 2>,
+      RAJA::statement::Lambda<0>
+    >
+  >
+
+>;
+
+#if defined(RAJA_ENABLE_OPENMP)
+using OpenMPKernel2DExecPols = camp::list< 
+    
+  RAJA::KernelPolicy<
+    RAJA::statement::For<0, RAJA::omp_parallel_for_exec,
+      RAJA::statement::For<1, RAJA::loop_exec,
+        RAJA::statement::Lambda<0>
+      >
+    >
+  >,
+
+  RAJA::KernelPolicy<
+    RAJA::statement::Collapse<RAJA::omp_parallel_collapse_exec, RAJA::ArgList<0, 1>,
+      RAJA::statement::Lambda<0>
+    >
+  >
+
+>;
+
+using OpenMPKernel3DExecPols = camp::list<
+   
+  RAJA::KernelPolicy<
+    RAJA::statement::For<0, RAJA::omp_parallel_for_exec,
+      RAJA::statement::For<1, RAJA::loop_exec,
+        RAJA::statement::For<2, RAJA::loop_exec,
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >
+  >,
+
+  RAJA::KernelPolicy<
+    RAJA::statement::Collapse<RAJA::omp_parallel_collapse_exec, RAJA::ArgList<0, 1, 2>,
+      RAJA::statement::Lambda<0>
+    >
+  >
+
+>;
+#endif  // if defined(RAJA_ENABLE_OPENMP)
+
+#if defined(RAJA_ENABLE_TBB)
+using TBBKernel2DExecPols = 
+camp::list< 
+
+  RAJA::KernelPolicy<
+    RAJA::statement::For<0, RAJA::tbb_for_exec,
+      RAJA::statement::For<1, RAJA::seq_exec,
+        RAJA::statement::Lambda<0>
+      >
+    >
+  >,
+
+  RAJA::KernelPolicy<
+    RAJA::statement::For<0, RAJA::tbb_for_static< >,
+      RAJA::statement::For<1, RAJA::seq_exec,
+        RAJA::statement::Lambda<0>
+      >
+    >
+  >
+
+>;
+
+using TBBKernel3DExecPols =
+camp::list<
+
+  RAJA::KernelPolicy<
+    RAJA::statement::For<0, RAJA::tbb_for_exec,
+      RAJA::statement::For<1, RAJA::seq_exec,
+        RAJA::statement::For<2, RAJA::seq_exec,
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >
+  >,
+
+  RAJA::KernelPolicy<
+    RAJA::statement::For<0, RAJA::tbb_for_static< >,
+      RAJA::statement::For<1, RAJA::seq_exec,
+        RAJA::statement::For<2, RAJA::seq_exec,
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >
+  >
+
+>;
+#endif  // if defined(RAJA_ENABLE_TBB)
+
+#if defined(RAJA_ENABLE_TARGET_OPENMP)
+using OpenMPTargetKernel2DExecPols =
+camp::list< 
+
+  RAJA::KernelPolicy<
+    RAJA::statement::Collapse<RAJA::omp_target_parallel_collapse_exec,
+                              RAJA::ArgList<0, 1>,
+      RAJA::statement::Lambda<0>
+    >
+  >
+
+>;
+
+using OpenMPTargetKernel3DExecPols =
+camp::list<
+
+  RAJA::KernelPolicy<
+    RAJA::statement::Collapse<RAJA::omp_target_parallel_collapse_exec,
+                              RAJA::ArgList<0, 1, 2>,
+      RAJA::statement::Lambda<0>
+    >
+  >
+
+>;
+#endif  // if defined(RAJA_ENABLE_TARGET_OPENMP)
+
+#if defined(RAJA_ENABLE_CUDA)
+using CudaKernel2DExecPols = 
+camp::list<
+
+  RAJA::KernelPolicy<
+    RAJA::statement::CudaKernel<
+      RAJA::statement::For<0, RAJA::cuda_thread_x_loop,
+        RAJA::statement::For<1, RAJA::cuda_thread_y_loop,
```

Comment:
should the inner loop use the x dimension?

Reply:
I don't think it really matters as the x/y/z block/thread dimensions aren't associated with data access (at least as I understand it). At any rate, these policy variations are here to make sure we don't have any underlying assumptions about execution patterns baked in.